lib/bridgetown-dato/builder.rb

```diff
@@ -3,8 +3,55 @@
 require "dato/site/client"
 require "dato/local/loader"
 require "dotenv"
+require "singleton"
 
 module BridgetownDato
+  class DatoLocal
+    include Singleton
+
+    attr_accessor :data
+
+    def loader(config)
+      @loader ||= begin
+        Bridgetown.logger.info("rage")
+        Dato::Local::Loader.new(
+          ::Dato::Site::Client.new(token),
+          config[:dato_config][:preview_mode]
+        )
+      end
+    end
+
+    def load(config)
+      loader(config)
+      @data ||= @loader.load
```

Comment:
Yes it is strange, but I don't know how to "initialize" a singleton.

I'm basically using this as a way to hide a global variable.  So technically if we just put this into a global variable and then had a class that you initialized that was the wrapper around that global variable - it would be one way.


I would have to do more research on the singleton module that comes with ruby.

Reply:
Yeah, I looked into it too and there doesn't seem to be a "nice" way to do it 😭 

This is all so tiny that it doesn't really change much, thankfully.